tests/aws/services/stepfunctions/v2/test_sfn_api.py

```diff
@@ -315,7 +315,58 @@ def test_list_sms(
         lst_resp_filter = [sm for sm in lst_resp["stateMachines"] if sm["name"] in sm_names]
         sfn_snapshot.match("lst_resp_del_filter", lst_resp_filter)
 
-    @markers.aws.needs_fixing
+    @markers.aws.validated
+    def test_start_execution_idempotent(
+        self, create_iam_role_for_sfn, create_state_machine, sfn_snapshot, aws_client
+    ):
+        snf_role_arn = create_iam_role_for_sfn()
+        sfn_snapshot.add_transformer(RegexTransformer(snf_role_arn, "snf_role_arn"))
+
+        sm_name: str = f"statemachine_{short_uid()}"
+        execution_name: str = f"execution_name_{short_uid()}"
+        definition = BaseTemplate.load_sfn_template(BaseTemplate.BASE_WAIT_10_SEC)
+        definition_str = json.dumps(definition)
+
+        creation_resp = create_state_machine(
+            name=sm_name, definition=definition_str, roleArn=snf_role_arn
+        )
+        sfn_snapshot.add_transformer(sfn_snapshot.transform.sfn_sm_create_arn(creation_resp, 0))
+        sfn_snapshot.match("creation_resp", creation_resp)
+        state_machine_arn = creation_resp["stateMachineArn"]
+
+        input_data = '{"body" : "data"}'
+        exec_resp = aws_client.stepfunctions.start_execution(
+            stateMachineArn=state_machine_arn, input=input_data, name=execution_name
+        )
+        sfn_snapshot.add_transformer(sfn_snapshot.transform.sfn_sm_exec_arn(exec_resp, 0))
+        sfn_snapshot.match("exec_resp", exec_resp)
+        execution_arn = exec_resp["executionArn"]
+
+        exec_resp_idempotent = aws_client.stepfunctions.start_execution(
+            stateMachineArn=state_machine_arn, input=input_data, name=execution_name
+        )
+        sfn_snapshot.add_transformer(
+            sfn_snapshot.transform.sfn_sm_exec_arn(exec_resp_idempotent, 0)
+        )
+        sfn_snapshot.match("exec_resp_idempotent", exec_resp_idempotent)
+
+        # Should fail because the execution has the same 'name' as another but a different 'input'.
+        with pytest.raises(Exception) as err:
+            aws_client.stepfunctions.start_execution(
+                stateMachineArn=state_machine_arn,
+                input='{"body" : "different-data"}',
+                name=execution_name,
+            )
+        sfn_snapshot.match("start_exec_already_exists", err.value.response)
+
+        await_execution_success(
```

Comment:
This test last 10 seconds as it waits for the wait state in the state machine to exit. You could use a [`waitForTaskToken`](https://docs.aws.amazon.com/step-functions/latest/dg/connect-to-resource.html#connect-wait-token) on an `sqs` queue `SendMessage` to have the running state machine wait until you can safely shut it down with a `SendTaskSuccess`. We have many tests for this service, you can take inspiration from those that use the `sqs_send_task_success_state_machine` fixture. This would also remove any flakiness should the second start command happen after the first execution terminated.

Reply:
I gave the `sqs_send_task_success_state_machine` fixture a look and I'm not sure it's too compatible with how these state machine and execution steps are run.

How about we increase the running time to 60 seconds (instead of 10) and terminate the execution after getting the execution already exists error (the final test case). Then we can block until the execution has terminated/finished. 

That way, the test will have an upper bound of 60 seconds (minimising flakiness) while also _technically_ being allowed to take less than 10 seconds -- making this faster in the average case.